exec/java-exec/src/main/java/org/apache/drill/exec/server/options/SystemOptionManager.java

```diff
@@ -251,7 +254,12 @@ public void setOption(final OptionValue value) {
     final String name = value.name.toLowerCase();
     final OptionValidator validator = getValidator(name);
 
-    validator.validate(value); // validate the option
+    /* If the validator depends on other options */
+    if (validator instanceof DependentTypeValidators) {
```

Comment:
It would be good to avoid doing an instanceof (although sometimes it becomes unavoidable).  Can you see what other changes are needed to achieve that ?  Seems like validate with a second parameter can be implemented by DependentTypeValidators only and other validators could throw an unsupported exception.